UITests/OktaScenarios.swift

```diff
@@ -0,0 +1,294 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import XCTest
+import OktaOidc
+
+class OktaScenarios: XCTestCase {
+    // Update these values along with your Plist config
+    var username = ProcessInfo.processInfo.environment["USERNAME"]!
+    var password = ProcessInfo.processInfo.environment["PASSWORD"]!
+    var issuer = ProcessInfo.processInfo.environment["ISSUER"]!
+    var redirectURI = ProcessInfo.processInfo.environment["REDIRECT_URI"]!
+    var logoutRedirectURI = ProcessInfo.processInfo.environment["LOGOUT_REDIRECT_URI"]!
+    var clientID = ProcessInfo.processInfo.environment["CLIENT_ID"]!
+    
+    let springboard = XCUIApplication(bundleIdentifier: "com.apple.springboard")
+    
+    private var browserContinueButton: XCUIElement {
+        springboard.buttons["Continue"]
+    }
+    
+    private var toolbarDoneButton: XCUIElement {
+        app.toolbars["Toolbar"].buttons["Done"]
+    }
+    
+    private var app: XCUIApplication!
+    
+    private var tokenTextView: XCUIElement {
+        app.textViews["tokenView"]
+    }
+    
+    private var signInButton: XCUIElement {
+        app.buttons["SignIn"]
+    }
+    
+    private var sessionTokenTextView: XCUIElement {
+        app.textViews["SessionTokenTextView"]
+    }
+    
+    private var getUserButton: XCUIElement {
+        app.buttons["GetUser"]
+    }
+    
+    private var introspectButton: XCUIElement {
+        app.buttons["Introspect"]
+    }
+    
+    private var revokeButton: XCUIElement {
+        app.buttons["Revoke"]
+    }
+    
+    private var clearButton: XCUIElement {
+        app.buttons["Clear"]
+    }
+    
+    private var signOutButton: XCUIElement {
+        app.buttons["SignOutOkta"]
+    }
+    
+    private var continueSpringboardButton: XCUIElement {
+        springboard.buttons["Continue"]
+    }
+    
+    private var authenticateButton: XCUIElement {
+        app.buttons["Authenticate"]
+    }
+    
+    // MARK: -
+    
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        
+        try XCTSkipIf(clientID.count == 0 || password.count == 0,
+                      "Cannot run UI tests without CLIENT_ID or PASSWORD environment variables")
+        
+        app = XCUIApplication()
+        app.launchEnvironment = [
+            "UITEST": "1",
+            "ISSUER": issuer,
+            "CLIENT_ID": clientID,
+            "REDIRECT_URI": redirectURI,
+            "LOGOUT_REDIRECT_URI" : logoutRedirectURI
+        ]
+        
+        continueAfterFailure = false
+        app.launch()
+        
+        if clearButton.exists {
+            clearButton.tap()
+        }
+    }
+    
+    override func tearDown() {
```

Comment:
You might want to remove this empty function, since it doesn't add anything at this point.

Reply:
Good input.